src/vault/vault_config.rs

```diff
@@ -0,0 +1,82 @@
+#![allow(non_snake_case)]
+
+use std::str::FromStr;
+use std::time::Duration;
+use mpc_utils::env::util::Util;
+
+#[derive(Clone, Debug)]
+pub enum AuthMethod {
+    Token,
+    Kubernetes,
+}
+
+impl FromStr for AuthMethod {
+    type Err = ();
+
+    fn from_str(str: &str) -> Result<Self, Self::Err> {
+        match str {
+            "Token" => Ok(AuthMethod::Token),
+            "Kubernetes" => Ok(AuthMethod::Kubernetes),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct VaultConfig {
+    pub auth_method: AuthMethod,
+    pub token: Option<String>,
+    pub role_name: Option<String>,
+    pub token_path: Option<String>,
+    pub address: String,
+    pub mount_path: String,
+    pub client_timeout: Duration,
+    pub healthcheck_file_path: String,
+    pub retry_count: u16,
+}
+
+impl VaultConfig {
+    pub fn loadEnv() -> VaultConfig {
+        let auth_method: AuthMethod = Util::loadOrDefault("VAULT_AUTH_METHOD", "Token")
+            .parse::<AuthMethod>()
+            .expect("Env variable: VAULT_AUTH_METHOD is not valid. Possible values: Token, Kubernetes");
```

Comment:
You should not panic from a library when something goes wrong. You should let the caller know the error let it handle it.